src/cclib/parser/daltonparser.py

```diff
@@ -737,20 +754,36 @@ def extract(self, inputfile, line):
         # @    Final gradient norm:           0.000003746706
         # ...
         #
+        if "Final HF energy" in line and not (hasattr(self, "mpenergies") or hasattr(self, "ccenergies")):
+            self.metadata["theory"] = "HF"
```

Comment:
Can we include all of these in a `methods` field? I don't think we benefit from any structure here, since the `metadata` dict will not be correlated with any other attributes. However, having a list of methods used you can still output CSX by checking what's inside.